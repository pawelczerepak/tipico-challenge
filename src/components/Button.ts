import styled from 'styled-components';

export default styled.button`
  flex: 1;
  margin: 0;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: .375rem .75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem .25rem .25rem 0.25rem;
  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out, box-shadow .15s ease-in-out;
  background-color: #fff;
  border: 1px solid #6c757d;
  &:not(:disabled) {
    cursor: pointer;
    color: #fff;
    background-color: #c8102e;
    border: 1px solid #c8102e;
  }
  &:disabled {
    opacity: .65;
  }
`;
