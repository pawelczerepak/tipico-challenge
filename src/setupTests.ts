import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
