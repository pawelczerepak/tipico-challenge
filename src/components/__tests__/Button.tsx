import React from 'react';
import '../../setupTests';
import Button from '../Button';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Button', () => {
    it('capture snapshot', () => {
        expect(toJson(shallow(<Button/>)))
            .toMatchSnapshot();
    });
});
