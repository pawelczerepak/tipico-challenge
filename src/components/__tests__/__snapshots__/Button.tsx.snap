// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button capture snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  flex: 1;
  margin: 0;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: .375rem .75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem .25rem .25rem 0.25rem;
  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out, box-shadow .15s ease-in-out;
  background-color: #fff;
  border: 1px solid #6c757d;
  &:not(:disabled) {
    cursor: pointer;
    color: #fff;
    background-color: #c8102e;
    border: 1px solid #c8102e;
  }
  &:disabled {
    opacity: .65;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
